pkg/apis/cnsoperator/config/cnsregistervolume_crd.yaml

```diff
@@ -33,10 +33,12 @@ spec:
             pvcName:
               description: Name of the PVC
               type: string
+              pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
             volumeID:
               description: VolumeID indicates an existing vsphere volume to be imported
                 into Project Pacific cluster
               type: string
+              pattern: '^[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$'
```

Comment:
I think we do not have FCD with capital case chars.

Reply:
@divyenpatel is there any documentation which says that? For now, I have used UUID format. 
@SandeepPissay can you confirm on this as well?

Reply:
I would suggest to not do FCD specific validation here as we need to support the same API for file volumes as well.

Reply:
makes sense.

Reply:
Can you explain this regex? What is allowed?

Reply:
> @divyenpatel is there any documentation which says that?

I do not have a link to documentation, but I have not seen volume ID in caps.

but I have one more question. Do we anticipate static file share volume provisioning using same CRD mechanism, then this pattern needs to be updated? as file share volume id is having some pre-fixes before UUID.

Reply:
regex says 
8 chars of (number, letters lower and caps) - 4 chars of (number, letters lower and caps) - 4 chars of (number, letters lower and caps) - 12 chars of (number, letters lower and caps)